### PR TITLE
nixos/services: invariant option docs MD conversions

### PR DESCRIPTION
**Pull request by pennae**
###### Description of changes

this is much the same as #182685, only bigger. it follows the same two rules: only touch `mkOption` arguments, and don't make changes that change the manual content.

the only tests we've done on this is to build the manual before and after the changes, then diffing the results: diff was empty.

another PR that could be done with the same rules would be to add `mdDoc` to all modules that *don't* have any docbook in their option docs. that might be a good for a transition period towards using MD for everything.

### CODE REVIEW

**Review by K900 — APPROVED** [review]
Diff LGTM. Let's get this going.

**Comment by SuperSandro2000** [comment]
I wouldn't mind if we split this up in batches of 30 or 50 or something like that.

File: nixos/modules/services/matrix/pantalaimon-options.nix [hunk 1]
```diff
@@ -60,9 +60,9 @@ with lib;
     extraSettings = mkOption {
       type = types.attrs;
       default = { };
-      description = ''
+      description = lib.mdDoc ''
         Extra configuration options. See
-        <link xlink:href="https://github.com/matrix-org/pantalaimon/blob/master/docs/man/pantalaimon.5.md">pantalaimon(5)</link>
+        [pantalaimon(5)](https://github.com/matrix-org/pantalaimon/blob/master/docs/man/pantalaimon.5.md)
```
Comment (jtojnar) [inline nixos/modules/services/matrix/pantalaimon-options.nix:65]: In the future, we might want to convert this into a manpage reference.

**Comment by pennae** [comment]
closing in favor of #183491

**Pull request closed without merging**